### DISCUSSION TITLE
Regelgeschwindigkeit

### DISCUSSION THREAD

**Discussion by firedp** [hoylabs/OpenDTU-OnBattery#908] [General]
Hallo und aus aktuellem Anlass. 
On Battery, aber ohne Batterie!
Ich sitze so vor on Battery und sehe wie mein ungeregeltes 300W BKW vollgas läuft und mein hm-600 weitere 600W liefern möchte, aber on Battery es schön regelt auf meinen Wunschwert, ich speise tatsächlich noch bewusst ein (eingestellt -168W).
Dann plötzlich geht der Staubsauger los und schnell bekommt der hm-600 die 100% Freigabe. Bis er allerdings seine Max von 612W bringt, dauert es etwas, OK, schlagartig von 300 auf 600 Watt will und macht er wohl nicht.
Umgekehrt aber auch, der Staubsauger geht aus, die Regelung fällt wieder auf den eingestellten -168W Wert für den hm-600, aber die Einpeiseleistung bleibt laut Hichi Tasmota Lesekopf bestimmt eine Minute oder länger oben bei über 700W, Invertergesamtleistung 918W.
Alle Werte fallen auch wieder langsam und schrittweise, obwohl das Mindest-Limit schon lange gesetzt wurde.
Klar möchte ich ungern und auch nur kurzfristig so viel einspeisen. Kann man die Regelung beschleunigen? Oder können die neuen hms Wechselrichter schneller regeln?
Aber ehrlich, ansonsten ist das ein wahnsinnig tolles Projekt und Programm. 
Und da es wahrscheinlich am BKW liegen wird, muss ich wahrscheinlich damit leben??

**Comment by schlimmchen** [comment]
Das scheint am Grid Profile zu hängen. Mein Inverter hat heute nach einem AC Verbindungsabbruch auch langsam geregelt. Aber im gewöhnlichen Betrieb schaltet der sehr schnell um. Bin gespannt zu erfahren, ob das jemand weiß, unter welchen Umständen der WR langsam zwischen Limits regelt. Vielleicht ist deine Frage im Discord besser aufgehoben, da ließt bestimmt jemand mit, der/die es genauer weiß.

**Comment by oldman4U** [comment]
Im Grid Profile kann man unterschiedliche Werte für die Regelung nach einem AC Abbruch und die Regelung selbst in %/sec einstellen. Dazu braucht man, so weit mir bekannt, aber einen Hoymiles Installer Zugang.

**Reply by firedp** [reply]
Danke, den Zugang habe ich natürlich nicht.
Aussteigen tut der WR auch nicht, lediglich dauert es schon einige Zeit bis er von 612 wieder runter auf 170W kommt. Er aktualisiert sein Bild alle 8-10 Sekunden und gefühlt ist er bei jedem neuen Abruf 80W runter, obwohl das reduzierte Limit schon feststeht. Definitiv sinkt die WR Leistung nicht abrupt von 612 auf 170W, da sind noch einige Schritte zwischen. 
Habe ja noch einen gleichen WR ein Jahr jünger und neuere Firmware drauf. Kann ja mal bei schönem Wetter wechseln, vielleicht regelt der andere schneller runter.

**Reply by spcqike** [reply]
Wenn du noch einen zweiten wechselrichter hast, vergleich doch bitte mal die Firmwares. Nicht, dass hoymiles da ein Update gemacht hat und der nun einfach anders und dafür langsamer regelt. 

Ich habe mal ein Video gesehen, ich glaube von Deye, wo der wechselrichter (erstmalig?) die AC Leistung „richtig“ begrenzt. Heißt, wenn man 200W einstellt regelt er tatsächlich 200W, und nicht einfach wie die hoymiles bisher 100W pro MPPT (für ein 2 MPPT Modell). Dadurch kompensiert die interne Regelung natürlich unterschiedliche verschattung der Eingänge, aber er war Sau langsam für meinen Geschmack. 

Wenn es wirklich der hoymiles ist, der einfach langsamer regelt, kann man da wohl nicht viel machen. Außer auf ein Firmware Update zu hoffen.

**Reply by firedp** [reply]
Der neuere Hoymiles hat eine andere Firmware, das hatte ich schon gesehen. Der erste ist von Anfang 2022 und den zweiten habe ich Anfang 2023 gekauft. 
Mal bei Gelegenheit testen, für den Wechsel muss ich auch auf meine Garage klettern. Ist was für wärmeres Wetter :-)

**Reply by spcqike** [reply]
> für den Wechsel muss ich auch auf meine Garage klettern. Ist was für wärmeres Wetter

auch deswegen hab ich angefangen für jedes Paneel einzelne Kabel bis in die Garage zu ziehen 🫣 damit ich später, wenn ich mal eine Batterie oder einen anderen WR anschließen will, nicht nochmal hoch muss 😄

**Reply by rainer667** [reply]
Hallo,
ich habe festgestellt das die Anzeige der Open DTU über http sehr verzögert kommt. Ich messe mit den Shellys da liegt das Ergebnis viel schneller an. Aber ich rede von 3-10 Sekunden nicht von Minuten.

**Reply by firedp** [reply]
Den Gedanken hatte ich auch kurz. Aber da im blauen Balken schon direkt der neue niedrige Wert angezeigt wird, warum soll dann die DTU den Hoymiles in seiner Regelgeschwindigkeit beeinflussen?  Das Limit scheint gesetzt, den Rest muss der Hoymiles machen. Zudem zeigt mir der Tasmota Lesekopf den aktuellen Bezugs- und Einspeisewert an.
Ich habe gerade mal Videos vom langsamen An- und Abstieg der Leistung gemacht. 

Wenn es einer sehen will, wie darf ich es liefern? Es sind rauf wie runter ca. 20-24 Watt pro 8 Sekunden.

**Reply by firedp** [reply]
Manchmal haue ich mir selbst ein Brett vor den Kopf.
Habe einfach den neueren hm-600 reguliert und der springt nicht in 24W Schritten sondern viel schneller.
Nun muss ich tauschen und gucken weil. An dem älteren HM600 hängen 2 JA Module a.345W und alles ca. 40 Meter entfernt auf der Wiese. Und der neuere HM600 hat derzeit nur ein Modul Trina bifacial mit 435W und steht 10 Meter entfernt auf der Garage.
Liegt es am WR, an den Modulen oder AC Leitungslänge, Entfernung?
Ich berichte bei Gelegenheit bzw. Nach Tausch der WRs.

Der ältere ist aus Kalenderwoche 16/2021 mit Firmware 1.0.8 und Hardware Teilenummer 269553685

der neuere 
aus 31/2022, Firmware 1.0.10, Teilenummer 269553683

**Comment by firedp** [comment]
> Den Gedanken hatte ich auch kurz. Aber da im blauen Balken schon direkt der neue niedrige Wert angezeigt wird, warum soll dann die DTU den Hoymiles in seiner Regelgeschwindigkeit beeinflussen? Das Limit scheint gesetzt, den Rest muss der Hoymiles machen. Zudem zeigt mir der Tasmota Lesekopf den aktuellen Bezugs- und Einspeisewert an. Ich habe gerade mal Videos vom langsamen An- und Abstieg der Leistung gemacht.
> 
> Wenn es einer sehen will, wie darf ich es liefern? Es sind rauf wie runter ca. 20-24 Watt pro 8 Sekunden.

**Comment by Manos1966** [comment]
@firedp poste mal bitte dein Tasmota Script,

**Comment by firedp** [comment]
> @firedp poste mal bitte dein Tasmota Script,

>D
>B
->sensor53 r
>M 1
+1,3,s,16,9600,ENERGY
1,77070100010800ff@1000,Gesamtverbrauch,kWh,Total,4
1,77070100020800ff@1000,Gesamteinspeisung,kWh,Supply,4
1,77070100100700ff@1,Actual Power,W,Power,0
1,77070100200700ff@1,Voltage L1,V,Voltage,1
1,77070100340700ff@1,Voltage L2,V,Voltage_L2,1
1,77070100480700ff@1,Voltage L3,V,Voltage_L3,1
1,770701001f0700ff@1,Current L1,A,Current,2
1,77070100330700ff@1,Current L2,A,Current_L2,2
1,77070100470700ff@1,Current L3,A,Current_L3,2
1,770701000e0700ff@1,Frequency,Hz,Freq,1
#

**Reply by Manos1966** [reply]
@firedp Danke!

Ich haette normalerweise dich gebeten, deine "Teleperiod" zu posten, aber es ist nicht relevant...

Also,

- lies mal https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/Power-Meter#important-for-tasmota-users
- Aendere das 1,77070100100700ff@1,Actual Power,W,Power,0 auf 1,77070100100700ff@1,Actual Power,W,Power,16

und probiere noch mal... 🤞

**Reply by firedp** [reply]
Was bewirkt das? Vom Gefühl aktualisiert sich Tasmota schneller.
Oder Plazebo-Effekt?😁

Den Wechselrichter beeindruckt das aber leider nicht in seiner Regelgeschwindigkeit. Immer schön in 20-30W Schritten alle 8 Sekunden

**Reply by Manos1966** [reply]
>Was bewirkt das? Vom Gefühl aktualisiert sich Tasmota schneller.

Ja, Tasmota sendet (erst) jetzt "Real-Time"

> Den Wechselrichter beeindruckt das aber leider nicht in seiner Regelgeschwindigkeit. Immer schön in 20-30W Schritten alle 8 Sekunden

Hmmm... 8 Sekunden Takt fuer 20-30W Schritte ist etwas lang

Du hast geschrieben: 
> die Einspeiseleistung bleibt laut Hichi Tasmota Lesekopf bestimmt eine Minute oder länger oben bei über 700W

Das sollte jetzt mit der Umstellung des Scripts schneller gehen und hoffentlich dadurch eine schnellere Reaktion des DPL und schnellere Reaktion des Wechselrichters. Vor diese Aenderung hing alles an die Teleperiod an. 
Hattest du Teleperiod nicht angepasst: Dann bekamst du eine Aktualisierung nach xx Sekunden, wie in der Teleperiod gespeichert ist (10Sek schnellstens, ueblich sind 300Sekunden) 😨 

> Alle Werte fallen auch wieder langsam und schrittweise, obwohl das Mindest-Limit schon lange gesetzt wurde.

Ich bin gespannt auf die Ergebnisse nach dem Umtausch der HM800. Vielleicht liegt es wirklich an die Firmware Version.
Mein HM1200 hat 1.0.16 und der HM1500 hat 1.0.18 daher kann ich nicht vergleichen. Da ich gerade Reaktionszeiten teste, finde ich beide reagieren sehr schnell.

**Reply by oldman4U** [reply]
Mein HM-800 hat FW 1.00.16. Hat Hoymiles da ein Update gemacht, oder hast du den so gekauft?

**Reply by oldman4U** [reply]
BTW die Funktion im Profil nennt sich Power Ramp Rate, ist unter Active Power Control zu finden und kann zwischen 0.33 und 100% pro Sekunde definiert werden.

**Reply by Manos1966** [reply]
Bei mir haben beide genau das gleiche (100%)

![Ramp Rate](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/1b2260f5-f33b-4c98-aac7-df268794e461)

**Reply by firedp** [reply]
Welche Funktion ist das?
Bei meinen 2 hm-600 s steht bei Power Ramp Rate 
0,5 Pn%/s 🤷🏼‍♂️🤷🏼‍♂️

**Reply by oldman4U** [reply]
Dann regeln die zwei mit 3W pro Sekunde.

**Reply by firedp** [reply]
Was ja dann theoretisch mein Problem erklärt.
8 Sekunden x 3 Watt sind 24W alle 8 Sekunden?

**Reply by Manos1966** [reply]
🤦‍♂️ 🤦‍♂️ 🤦‍♂️ 🤦‍♂️ 🤦‍♂️ 🤦‍♂️ 

Laesst sich das Power Ramp Rate aendern???

@oldman4U how do you calculate the Power Ramp Rate 0,5 Pn%/s ?

**Reply by oldman4U** [reply]
Hoymiles Installer account.

0,5% von 600W Inverterleistung sind 3W pro Sekunde.

**Reply by firedp** [reply]
Das mit dem Ändern wäre ja gut. Aber sollte der neuere hm-600 wirklich schneller sein, wäre die Erklärung PPR ja fast hinfällig.

**Comment by firedp** [comment]
> > Was bewirkt das? Vom Gefühl aktualisiert sich Tasmota schneller.
> 
> Ja, Tasmota sendet (erst) jetzt "Real-Time"
> 
> > Den Wechselrichter beeindruckt das aber leider nicht in seiner Regelgeschwindigkeit. Immer schön in 20-30W Schritten alle 8 Sekunden
> 
> Hmmm... 8 Sekunden Takt fuer 20-30W Schritte ist etwas lang
> 
> Du hast geschrieben:
> 
> > die Einspeiseleistung bleibt laut Hichi Tasmota Lesekopf bestimmt eine Minute oder länger oben bei über 700W
> 
> Das sollte jetzt mit der Umstellung des Scripts schneller gehen und hoffentlich dadurch eine schnellere Reaktion des DPL und schnellere Reaktion des Wechselrichters. Vor diese Aenderung hing alles an die Teleperiod an. Hattest du Teleperiod nicht angepasst: Dann bekamst du eine Aktualisierung nach xx Sekunden, wie in der Teleperiod gespeichert ist (10Sek schnellstens, ueblich sind 300Sekunden) 😨
> 
> > Alle Werte fallen auch wieder langsam und schrittweise, obwohl das Mindest-Limit schon lange gesetzt wurde.
> 
> Ich bin gespannt auf die Ergebnisse nach dem Umtausch der HM800. Vielleicht liegt es wirklich an die Firmware Version. Mein HM1200 hat 1.0.16 und der HM1500 hat 1.0.18 daher kann ich nicht vergleichen. Da ich gerade Reaktionszeiten teste, finde ich beide reagieren sehr schnell.

Bei der Hichi/Tasmota-Sache haben wir uns missverstanden, der war schon vorher schnell. Ich wollte damit ausdrücken, dass ich den hohen Einspeisewert lange ablese, bzw. das langsame Sinken beobachte, was aber vom Hoymiles verursacht wird.
Bzw. Das openDTU on Battery braucht selber bei der Abfrage von den zwei Wechselrichtern mindestens 7-10 Sekunden, selbst wenn ich eine schnellere Aktualisierung einstelle.

**Reply by Manos1966** [reply]
Neee, ich habe dich korrekt verstanden.

Es ist so: Teleperiod bei Tasmota hat bestenfalls 10 Sekunden. Es ist aber ein Parameter, dass nur für die Übermittlung MQTT Nachrichten wichtig ist.
Ich gehe davon aus, deine Abfrage ist HTTP+JSON (kein MQTT), es sollte also keine Rolle spielen was fuer Wert, Teleperiod hat.
Der +16 "Trick" sollte also auch nur MQTT-Relevant sein. Aber manchmal hilft es trotzdem.

Ich bin gespannt zu erfahren, ob dein Power Ramp Rate andere Werte hat, als bei mir.

**Reply by firedp** [reply]
Du hast es ja weiter oben wohl schon gelesen 0,5 Power Ramp Rate.
Und richtig, HTTP+JSON

**Reply by oldman4U** [reply]
Wo die Inverter auch immer her sind. Derjenige soll das default TOR installieren und gut ist es. Hier ist ein verändertes installiert und keiner weiß was sonst noch verstellt ist. Da kann man auch die Inselerkennung abschalten, was äußerst gefährlich sein könnte.

**Reply by firedp** [reply]
Also meine hm-600 sind neu und original gekauft. Von unterschiedlichen Anbietern mit einem Abstand von ca. einem Jahr. Denke bei meinen wurde nichts verstellt.

**Reply by firedp** [reply]
Ich bin bei dem Wetter doch raus und habe die Wechselrichter ausgetauscht.
Und tatsächlich, der neuere regelt "knallhart" ohne Zwischenschritte. Also liegt es wahrscheinlich am WR. Die beiden hm-600 haben ja unterschiedliche Nummern bei der Hardware (also nicht die Seriennummer ;-) ) und bei der Firmware.
Bleibt jetzt mal so. der alte bleibt ungeregelt mit dem 435er Trina und liefert an dem Eingang regelmäßig oft 306 W und der neuere bedient jetzt die zwei 345er JA Solar und wird von on Battery nach bedarf geregelt.

![Unbenannt2](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/126169749/8623ca5c-0116-491d-84d2-4a775c06658d)

**Reply by Manos1966** [reply]
@firedp  RESPEKT! (es schneit bei mir...)
Andererseits, optimale (dunkle) Verhaeltnisse um Paneele umzustecken 😉 

Bleibt leider weiterhin der grosse Unbekannte: "Bei meinen 2 hm-600 s steht bei Power Ramp Rate 0,5 Pn%/s"
Wenn BEIDE Power Ramp Rate 0,5 Pn%/s haben, dann ist diese Einstellung nicht der entscheidende Punkt.

Ich will hoffen, dein langsamer Wechselrichter hat die Firmware 1.0.8, so kann man hoffen dass ein Firmware Upgrade die erwartete Verbesserung bringen wird.

Ansonsten, eine gute Gelegenheit zu testen, wie gut die Hoymiles Garantie ist. 🤞 🤞 🤞

**Reply by spcqike** [reply]
> so kann man hoffen dass ein Firmware Upgrade die erwartete Verbesserung bringen wird.

es könnte auch die Firmware 1.0.10 ein bug haben. Und durch 1.0.11 gefixxt. Wenn bei beiden 0.5%/s hinterlegt ist, vielleicht bugged eine Firmware rum und regelt immer voll, egal was gesetzt ist? 

Wie auch immer, so richtig rund läuft es nicht. Einer von beiden scheint ein Problem zu haben. Und vielleicht haben beide ein falsches Profil.

**Reply by firedp** [reply]
Der langsame ist der 1.0.8. War mein erster WR im Jahr 2022 gekauft. Das lahme Verhalten wäre eigentlich nie aufgefallen und er bringt ja bei Sonne immer seine 612W und fällt nie aus oder zeigt Fehler. Da ich keinen Hoymiles Account habe, werde ich auch nicht updaten (können).
Somit bin ich gerade zufrieden mit dem Ergebnis und bremse mich mit weiteren Investitionen in Module oder anderen/mehr Wechselrichter. Meine aktuell möglichen 918W werden so selten benötigt und on Battery kümmert sich zuverlässig um eine sehr kleine 3-stellige Überschusseinspeisung ins Netz, ich gebe ja gerne, nur nicht zu viel ;-)

**Reply by oldman4U** [reply]
Da nicht bekannt ist was sonst noch alles verstellt ist würde ich dringend raten das default TOR aufspielen zu lassen.

**Reply by firedp** [reply]
Ich lasse es jetzt so und beobachte. Derzeit diagnostiziere ich ein „normales“ Verhalten gemäß Firmware 1.0.8
Jetzt eine originale DTU kaufen um rauszufinden ob es behebbar ist, macht mich zwar neugierig, aber ist es aktuell nicht wert.
Da eigentlich niemand etwas verstellt haben kann, mache ich mir keinen Kopf. Ohne on Battery und meine zufälligen Beobachtungen am offenen Bildschirm wäre es gar nicht aufgefallen, bzw. hätte ich es sogar als normales Verhalten abgetan.

Jedenfalls habe ich wieder gelernt und danke Euch für die hilfreichen Kommentare und Geduld.

**Comment by firedp** [comment]
> @firedp Danke!
> 
> Ich haette normalerweise dich gebeten, deine "Teleperiod" zu posten, aber es ist nicht relevant...
> 
> Also,
> 
> * lies mal https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/Power-Meter#important-for-tasmota-users
> * Aendere das 1,77070100100700ff@1,Actual Power,W,Power,0 auf 1,77070100100700ff@1,Actual Power,W,Power,16
> 
> und probiere noch mal... 🤞



> BTW die Funktion im Profil nennt sich Power Ramp Rate, ist unter Active Power Control zu finden und kann zwischen 0.33 und 100% pro Sekunde definiert werden.



> Wo die Inverter auch immer her sind. Derjenige soll das default TOR installieren und gut ist es. Hier ist ein verändertes installiert und keiner weiß was sonst noch verstellt ist. Da kann man auch die Inselerkennung abschalten, was äußerst gefährlich sein könnte.
